internal/persistence/persistence.go

```
@@ -279,7 +306,7 @@ func (s *Storage) LoadShard(shardID uint32, engine *storage.Engine) error {
 		var n int
 		n, err = io.ReadFull(f, header)
 		if err != nil {
-			if err == io.EOF || err == io.ErrUnexpectedEOF {
+			if err == io.EOF || errors.Is(io.ErrUnexpectedEOF, err) {
```

golangci-lint (2.12.2)

[error] SA1032: arguments have the wrong order

(staticcheck)